src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -547,27 +548,69 @@ void Terminal::SelectHyperlink(const SearchDirection dir)
                 searchArea = Viewport::FromDimensions(searchStart, { searchEnd.x + 1, searchEnd.y + 1 });
             }
         }
+    }
 
-        // 1.C) Nothing was found. Bail!
-        if (!result.has_value())
+    // 2. We found a hyperlink from the pattern tree. Look for embedded hyperlinks too!
+    // Use the result (if one was found) to narrow down the search.
+    searchStart = dir == SearchDirection::Forward ?
+                      _selection->start :
+                      (result ? result->second : bufferSize.Origin());
+    searchEnd = dir == SearchDirection::Forward ?
+                    (result ? result->first : buffer.GetLastNonSpaceCharacter()) :
+                    _selection->start;
+    auto iter = buffer.GetCellDataAt(dir == SearchDirection::Forward ? searchStart : searchEnd);
+    while (dir == SearchDirection::Forward ? iter.Pos() < searchEnd : iter.Pos() > searchStart)
+    {
+        // Don't let us select the same hyperlink again
+        if (iter.Pos() < _selection->start || iter.Pos() > _selection->end)
         {
-            return;
+            if (auto attr = iter->TextAttr(); attr.IsHyperlink())
+            {
+                // Found an embedded hyperlink!
+                const auto hyperlinkId = attr.GetHyperlinkId();
+
+                // Expand the start to include the entire hyperlink
+                TextBufferCellIterator hyperlinkStartIter{ buffer, iter.Pos() };
+                while (attr.IsHyperlink() && attr.GetHyperlinkId() == hyperlinkId)

```

Comment:
If the hyperlink starts at the origin of the search, couldn't this accidentally leave the boundaries of it?

Reply:
Good catch. Fixed!

This also found another bug that I fixed in the latest commit. If my selection was at the exclusive end position of the row, we would crash when attempting to `buffer.GetCellDataAt()` it. Now I clamp just to be safe